compiler/src/iree/compiler/Dialect/Flow/Transforms/test/dispatch_linalg_on_tensors.mlir

```diff
@@ -118,7 +118,7 @@ util.func public @fuse_matmul_with_fill(%A : tensor<?x?xf32>, %B : tensor<?x?xf3
 //   CHECK-DAG:     %[[ARG0_DIM1:.+]] = tensor.dim %[[ARG0]], %[[C1]]
 //   CHECK-DAG:     %[[ARG1_DIM0:.+]] = tensor.dim %[[ARG1]], %[[C0]]
 //   CHECK-DAG:     %[[ARG1_DIM1:.+]] = tensor.dim %[[ARG1]], %[[C1]]
-//  CHECK-NEXT:     flow.dispatch.workgroups[%[[ARG0_DIM0]], %[[ARG0_DIM1]], %[[ARG1_DIM0]], %[[ARG1_DIM1]]]
+//  CHECK-NEXT:     flow.dispatch.workgroups[%[[ARG0_DIM0]], %[[ARG1_DIM1]], %[[ARG0_DIM1]], %[[ARG1_DIM0]]]
```

Comment:
Moving canonicalization patterns out of the pass and instead running a the mlir canonicalization pass caused the order of the workload args to change. Is this a problem? and if not, is there a more resilient way to write filechecks

Reply:
Oh, that shouldnt happen. This is like a function ABI. So these need to remain the same.

Reply:
Actually it is consistent. Can you see why this happened?

Reply:
I wasn't invoking canonicalization before `MaterializeDefaultWorkgroupCountRegion` during the lit tests. The previous version applied `ClosureOptimizationPattern` which messes with the ordering of the args. Registering and using the new pipeline should fix this issue